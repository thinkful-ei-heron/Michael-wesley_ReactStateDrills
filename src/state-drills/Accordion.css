body{
  background-color: #454C5E;
  padding: 3rem;
}

ul{
  list-style: none;
  margin-left:0;
  padding-left:0;
}

.btn{
  width: 40%;
  min-width: 10rem;
  max-width: 23rem;
  height: 125%;
  font-size: 2rem;
  margin: 10px;
  border-radius: 25px;
  background-color: #F5ECD4;
  color: #454C5E;
  box-shadow: 5px 5px;
}

.btn:focus{
  outline: none;
  box-shadow: 0 0 10px 5px #CF5F38;
}

p{
  color: #F5ECD4;
  font-size: 1.5rem;
}