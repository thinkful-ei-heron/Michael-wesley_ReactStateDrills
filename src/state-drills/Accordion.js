import React, { Component } from 'react'

//props: sections = [{title: str, content: str}, {title: str, content: str}, ...]

export class Accordion extends Component {

  state = {
    expandedSection: null
  }

  componentDidMount() {

  }

  /*for each non-expanded section:
<li class='section'>
  <button type='button' onClick={this.foo}>{sectionX.title}</button>
</li>

  for each expanded section
<li class='section'>
  <button type='button' onClick={this.foo}>{sectionX.title}</button>
  <p>{sectionX.content}</p>
</li>
  */

  closeSection() {
    this.setState({
      expandedSection: null
    })
  }

  openSection(idx) {
    this.setState({
      expandedSection: idx
    })
  }

  renderSection(section, idx){
    if(idx === this.state.expandedSection){
      return (
        <li className='section expanded' key={'section' + idx}>
          <button type='button' className='btn expanded' onClick={() => this.closeSection()}>{section.title}</button>
          <p>{section.content}</p>
        </li>
      )
    } else {
      return (
        <li className='section'  key={'section' + idx}>
          <button type='button' className='btn' onClick={() => this.openSection(idx)}>{section.title}</button>
        </li>
      )
    }
  }

  render() {
    if(this.props.sections === undefined) {
      return (
        <div>
          <ul>
            <li key="0"></li>
          </ul>
        </div>
      )
    }

    return (
      <div>
        <ul>
        {this.props.sections.map((section, idx) => this.renderSection(section, idx))}
        </ul>
      </div>
    )
  }
}

export default Accordion
