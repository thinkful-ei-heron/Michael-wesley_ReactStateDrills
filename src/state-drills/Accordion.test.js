import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Accordion';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe('Accordion test', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Accordion/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders as expected when sections not supplied', ()=>{
    const wrapper = shallow(<Accordion/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders no sections active by default', () => {
    expect(renderer
      .create(<Accordion sections={sections} />)
      .toJSON())
      .toMatchSnapshot()
  })
  it('opens a section on click', () =>{
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('only opens the most recently clicked section when multiple sections have been clicked', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})
