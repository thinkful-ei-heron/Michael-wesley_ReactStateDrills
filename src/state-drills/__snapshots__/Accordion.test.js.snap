// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion test only opens the most recently clicked section when multiple sections have been clicked 1`] = `
<div>
  <ul>
    <li
      className="section"
      key="section0"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
    </li>
    <li
      className="section"
      key="section1"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
    </li>
    <li
      className="section expanded"
      key="section2"
    >
      <button
        className="btn expanded"
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
      <p>
        Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
      </p>
    </li>
  </ul>
</div>
`;

exports[`Accordion test opens a section on click 1`] = `
<div>
  <ul>
    <li
      className="section"
      key="section0"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
    </li>
    <li
      className="section expanded"
      key="section1"
    >
      <button
        className="btn expanded"
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      className="section"
      key="section2"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion test renders as expected when sections not supplied 1`] = `
<div>
  <ul>
    <li
      key="0"
    />
  </ul>
</div>
`;

exports[`Accordion test renders no sections active by default 1`] = `
<div>
  <ul>
    <li
      className="section"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
    </li>
    <li
      className="section"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
    </li>
    <li
      className="section"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion test renders without crashing 1`] = `
<div>
  <ul>
    <li
      key="0"
    />
  </ul>
</div>
`;
